lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,523 @@
+            )
+
+        # validate the email field
+        try:
+            valid = validate_email(emailID)
+            email = valid.email
+        except EmailNotValidError as e:
+            self.logger.error(f"Register User: invalid email {emailID}")
+            abort(400, message=f"Invalid email: {emailID}")
+
+        # check if user already exist
+        user = UserModel.query.filter_by(username=user_data.get("username")).first()
+        if not user:
+            try:
+                user = UserModel(
+                    bcrypt=self.bcrypt,
+                    bcrypt_log_rounds=self.bcrypt_log_rounds,
+                    username=username.lower(),
+                    password=password,
+                    firstName=firstName,
+                    lastName=lastName,
+                    email=email,
+                )
+
+                # insert the user
+                self.db_session.add(user)
+                self.db_session.commit()
+                self.logger.info(
+                    f"New user registered, Name: {firstName} {lastName} username: {username}, email: {email}"
+                )
+
+                # generate the auth token
+                auth_token = user.encode_auth_token(
+                    self.token_expire_duration, user.id, self.logger
+                )
+                if not auth_token:
+                    self.logger.info(
+                        f"User registered but unable to encode the auth token. username {username}"
+                    )
+                    return make_response(
+                        "{'message':'INTERNAL ERROR, unable to encode the auth token, please try logging in}",
+                        500,
+                    )
+
+                self.logger.info(auth_token.decode())
+                response_object = {
+                    "status": "success",
+                    "message": "Successfully registered.",
+                    "auth_token": auth_token.decode(),
+                }
+                response = jsonify(response_object)
+                response.status_code = 201
+                return make_response(response, 201)
+                # return Response(response_object, status=201, mimetype="application/json")
+
+            except (
+                SQLAlchemyError,
+                DataError,
+                DatabaseError,
+                DBAPIError,
+                CompileError,
+            ) as e:
+                self.logger.error(
+                    f"Some SQLAlchemy error occurred while registering a user {e.__dict__['orig']}"
+                )
+                abort(500, message="INTERNAL ERROR, Please try again.")
+
+            except Exception as e:
+                self.logger.error(f"Exception while registering a user {e}")
+                abort(500, message="INTERNAL ERROR, Please try again.")
+        else:
+            abort(409, message="User already exists. Please Log in.")
```

Comment:
Stylistically, instead of indenting 50+ lines of code, perhaps we could just check for the `409` condition at line 148 and invoke `abort(409)`?

Also, perhaps we could use a more common wording, "A user with that name already exists.", without suggesting an action.  They may want to create a user and not try to log into the existing user.